### PR TITLE
80 flake8 linting

### PR DESCRIPTION
**Pull request by wwymak**
part of #80 
- added flake8 config 
- run linter on ci test
- fix all the code style errors flake8 found
- added flake8 and autopep8 to dev dependencies

### CODE REVIEW

File: flood_forecast/train_model.py [hunk 1]
```diff
@@ -1,8 +0,0 @@
-import argparse
```
Comment (wwymak) [inline flood_forecast/train_model.py:1]: I am assuming you don't want a file that only does 1+1 so I deleted it...

File: .flake8 [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[flake8]
+max_line_length=120
+ignore=E305,W504
```
Comment (wwymak) [inline .flake8:3]: E305 is https://www.flake8rules.com/rules/E305.html and w504 is https://www.flake8rules.com/rules/W504.html

File: flood_forecast/evaluator.py [hunk 3]
```diff
@@ -106,18 +121,32 @@ def infer_on_torch_model(
             or no columns if num_prediction_samples is None
     """
     device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
-    if type(datetime_start) == str:
+    if isinstance(datetime_start, str):
```
Comment (isaacmg) [inline flood_forecast/evaluator.py:124]: Are instances better?

File: .flake8 (hunk 2 above)
Comment (isaacmg) [reply .flake8:3]: Okay

File: setup.py [hunk 4]
```diff
@@ -8,11 +8,23 @@
     with open(requirementPath) as f:
         install_requires = f.read().splitlines()
 
+dev_requirements = [
+    'autopep8',
+    'flake8'
```
Comment (isaacmg) [inline setup.py:13]: Is there a difference?

File: flood_forecast/train_model.py (hunk 1 above)
Comment (isaacmg) [reply flood_forecast/train_model.py:1]: Yeah that can be removed.

File: setup.py (hunk 4 above)
Comment (wwymak) [reply setup.py:13]: autopep8 lets you do automatic fixing of lint errors e.g. `autopep8 --in-place --aggressive --aggressive --recursive .` whereas flake8 only tells you what errors you have

File: flood_forecast/evaluator.py (hunk 3 above)
Comment (wwymak) [reply flood_forecast/evaluator.py:124]: in general isinstance is a bit nicer than type(x) == type(y) since isinstance takes care of subclasses etc.

**Pull request merged**